plugin/trino-google-sheets/src/test/java/io/trino/plugin/google/sheets/TestGoogleSheets.java

```diff
@@ -13,29 +13,123 @@
  */
 package io.trino.plugin.google.sheets;
 
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.SheetsScopes;
+import com.google.api.services.sheets.v4.model.Sheet;
+import com.google.api.services.sheets.v4.model.SheetProperties;
+import com.google.api.services.sheets.v4.model.Spreadsheet;
+import com.google.api.services.sheets.v4.model.SpreadsheetProperties;
+import com.google.api.services.sheets.v4.model.UpdateValuesResponse;
+import com.google.api.services.sheets.v4.model.ValueRange;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import io.trino.testing.AbstractTestQueryFramework;
 import io.trino.testing.QueryRunner;
+import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import static com.google.api.client.googleapis.javanet.GoogleNetHttpTransport.newTrustedTransport;
 import static io.trino.plugin.google.sheets.SheetsQueryRunner.createSheetsQueryRunner;
 import static io.trino.plugin.google.sheets.TestSheetsPlugin.DATA_SHEET_ID;
-import static io.trino.plugin.google.sheets.TestSheetsPlugin.TEST_METADATA_SHEET_ID;
+import static io.trino.plugin.google.sheets.TestSheetsPlugin.getTestCredentialsPath;
+import static java.lang.Math.toIntExact;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 
 public class TestGoogleSheets
         extends AbstractTestQueryFramework
 {
+    private static final String APPLICATION_NAME = "trino google sheets integration test";
+    private static final String TEST_SPREADSHEET_NAME = "Trino integration test";
+
+    private Sheets sheetsService;
+    private String spreadsheetId;
+
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        return createSheetsQueryRunner(
-                ImmutableMap.of(),
-                ImmutableMap.of(
-                        "gsheets.read-timeout", "1m",
-                        "gsheets.metadata-sheet-id", TEST_METADATA_SHEET_ID));
+        sheetsService = getSheetsService();
+        spreadsheetId = createSpreadsheetWithTestdata();
+        return createSheetsQueryRunner(ImmutableMap.of(), ImmutableMap.of(
+                "gsheets.metadata-sheet-id", spreadsheetId + "#Metadata",
+                "gsheets.connection-timeout", "1m",
+                "gsheets.read-timeout", "1m",
+                "gsheets.write-timeout", "1m"));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
```

Comment:
I would remove this method and instead wrote a comment that this test is always creating new file.

Is it possible that Google Sheets has too many files and creation of new file would fail? 

If there are many files, is it possible that this service would be slow to respond.

Reply:
I am not aware of any of these restrictions, the goal was to "be a good citizen" by removing sheets that are not used any more.
I can remove the deletion part, but a short confirmation would be appreciated before removing the code.

Reply:
Please remove it. Let's keep the short comment next to variable declaration. It will be enough.

I believe that the current approach is worse, because the method is defined so one can think we clear things but then in the method body it in the code the removal does not happen. So not having a method that supposed to clean is clear message that there is no spreadsheet removal.

Reply:
Alright, removed it and kept the comment